routing/turns.hpp

```diff
@@ -18,36 +18,43 @@ using TEdgeWeight = double;
 
 /// \brief Unique identification for a road edge between two junctions (joints).
 /// In case of OSRM it's NodeID and in case of RoadGraph (IndexGraph)
-/// it's mwm id, feature id, segment id and direction.
+/// it's mwm id, feature id, range of segment ids [|m_startSegId|, |m_endSegId|) and direction.
 struct UniNodeId
 {
   enum class Type
   {
-    Osrm,
-    Mwm,
+    Osrm, // It's an OSRM node id so only |m_type| and |m_nodeId| are valid.
+    Mwm,  // It's a node for A* router so |m_nodeId| is not valid.
   };
 
   UniNodeId(Type type) : m_type(type) {}
-  UniNodeId(FeatureID const & featureId, uint32_t segId, bool forward)
-    : m_type(Type::Mwm), m_featureId(featureId), m_segId(segId), m_forward(forward)
+  UniNodeId(FeatureID const & featureId, uint32_t startSegId, uint32_t endSegId, bool forward)
+    : m_type(Type::Mwm)
+    , m_featureId(featureId)
+    , m_startSegId(startSegId)
+    , m_endSegId(endSegId)
+    , m_forward(forward)
   {
+    if (!m_featureId.IsValid())
+      m_nodeId = nextFakeId++;
```

Comment:
What about thread-safety? Is it guaranteed that instances of this class are always instantiated on the same thread, or with necessary fences?